include/fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp

```diff
@@ -0,0 +1,2876 @@
+     *        Some TypeIdentifiers do not involve computing the Hash of any TypeObject. These are called
+     *        Fully-descriptive TypeIdentifiers because they fully describe the Type. These are:
+     *        - The TypeIdentifiers for Primitive and String types.
+     *        - The TypeIdentifiers of plain collections where the element (and key) TypeIdentifier is a fully
+     *          descriptive TypeIdentifier. They are recognized by the contained PlainCollectionHeader having
+     *          EquivalenceKind set to EK_BOTH.
+     *
+     * @param[in] type_identifier TypeIdentifier to check.
+     * @return true if the given TypeIdentifier is fully-descriptive. false otherwise.
+     */
+    static bool is_fully_descriptive_type_identifier(
+            const TypeIdentifier& type_identifier);
+
+    /**
+     * @brief Check if a given TypeIdentifier is direct hash.
+     *        XTypes v1.3 Clause 7.3.4.6.3
+     *        These are HASH TypeIdentifiers with discriminator EK_MINIMAL, EK_COMPLETE or TI_STRONG_COMPONENT.
+     *
+     * @param[in] type_identifier TypeIdentifier to check.
+     * @return true if the given TypeIdentifier is direct hash. false otherwise.
+     */
+    static bool is_direct_hash_type_identifier(
+            const TypeIdentifier& type_identifier);
+
+    /**
+     * @brief Check if a given TypeIdentifier is indirect hash.
+     *        XTypes v1.3 Clause 7.3.4.6.4
+     *        These are the TypeIdentifiers for plain collections with the element type identified using a HASH
+     *        TypeIdentifier. They are distinguished by:
+     *        1. Having discriminator TI_PLAIN_SEQUENCE_SMALL, TI_PLAIN_SEQUENCE_LARGE, TI_PLAIN_ARRAY_SMALL,
+     *           TI_PLAIN_ARRAY_LARGE, TI_PLAIN_MAP_SMALL, or TI_PLAIN_MAP_LARGE.
+     *        2. Having the contained PlainCollectionHeader with EquivalenceKind EK_MINIMAL or EK_COMPLETE.
+     *
+     * @param[in] type_identifier TypeIdentifier to check.
+     * @return true if the given TypeIdentifier is indirect hash. false otherwise.
+     */
+    static bool is_indirect_hash_type_identifier(
+            const TypeIdentifier& type_identifier);
+
+    /*************** Consistency methods (Debug) ***************************/
+
+    /**
+     * TypeObjectHashId is always consistent. Default constructor already sets the discriminator to one valid value.
+     * Union setters prevent changing the discriminator value without setting the corresponding union member.
+     */
+
+    /**
+     * @brief Check bound consistency: must be different from 0.
+     *
+     * @tparam Either SBound or LBound.
+     * @param[in] bound Bound to be checked.
+     * @exception eprosima::fastdds::dds::xtypes1_3::InvalidArgumentError exception if the given bound is not
+     *            consistent.
+     */
+    template<typename T>
+    static void bound_consistency(
+            T bound)
+    {
+        if (INVALID_LBOUND == bound)
```

Comment:
I doubt that the values change at some point. It does not make sense a collection with no elements (which is the invalid number 0). The difference between SBound and LBound is the size. SBound is defined as an octet (1 byte) whereas LBound is an unsigned long (4 bytes). I am comparing with the INVALID_LBOUND value, because it could be that the LBound has all zeros in the SBound positions, but its not zero as the more significant bytes could be different from 0. However, using the 4 bytes 0, INVALID_LBOUND, the comparison with SBound is valid too.